src/emulator/modules/SceLibKernel/SceLibRng.cpp

```diff
@@ -17,8 +17,56 @@
 
 #include "SceLibRng.h"
 
-EXPORT(int, sceKernelGetRandomNumber) {
-    return unimplemented("sceKernelGetRandomNumber");
+#if defined(__APPLE__)
+
+#include <sys/random.h>
+
+#elif defined(WIN32)
+
+// copy from vmware/open-vm-tools
+#include <windows.h>
+#include <wincrypt.h>
+
+void read_random(void *output, size_t size) {
+    HCRYPTPROV csp;
+
+    if (size != (DWORD)size) {
+        return;
+    }
+
+    if (CryptAcquireContext(&csp, NULL, NULL, PROV_RSA_FULL,
+                            CRYPT_VERIFYCONTEXT) == false) {
+        return;
+    }
+
+    if (CryptGenRandom(csp, (DWORD)size, buffer) == false) {
+        CryptReleaseContext(csp, 0);
+        return;
+    }
+
+    if (CryptReleaseContext(csp, 0) == false) {
+        return;
+    }
+
+    return size;
+}
+
+#else
+
+#include <sys/random.h>
+
+void read_random(void *output, unsigned int size) {
+    getrandom(output, size, GRND_RANDOM);
+}
+
+#endif
+
+EXPORT(int, sceKernelGetRandomNumber, void *output, unsigned int size) {
+    if (size > 64) {
+        return -1;
```

Comment:
This should be replaced with proper erroring using RET_ERROR

Reply:
use `error`, my PR with `RET_ERROR` isn't merged yet